base/task.jl

```diff
@@ -208,6 +225,7 @@ function produce(v)
         else
             schedule_and_wait(t, v)
         end
+        ct = current_task() # When a task is copied, ct should be updated to new task ID.
```

Comment:
I guess this is one reason it's hard to copy a running task since the current task can basically change on every task switches.